### ISSUE TITLE
[BUG]: Export Project sometimes does nothing after a failed export.

### ISSUE THREAD

**Issue by Kangie** [Zettlr/Zettlr#1227]
## Description
Export Project _sometimes_ does nothing after a failed export, lasting until the application is restarted.

## Reproducing

- Create a project with a few documents and break the pandoc export (an undefined control sequence does nicely.)
- Dismiss the error, fix the breakage, and attempt to re-run the export. 
- No folders are generated within the project directory for the export, and the export does not run.

Once this bug is encountered, it occurs consistently until the application has been restarted - It will then work until the bug is encountered again.

Note that not all failed exports cause this issue - I can export 2-3 times while fixing up pandoc errors before it happens.

## Expected behaviour
Zettlr should either export successfully, or error, not silently fail.

**Screenshots**
Can't provide any screenshots, and there's nothing in the built-in zettlr web engine (not sure which one you're using!) console - is there a good way to get console output from Zettlr so that I can provide more reproducible / informative bugs?

## Platform
 - OS and version: Windows 10 (20.04)
 - Zettlr Version: 1.7.5
 - Screen Resolution: 1920x1080

**Comment by Kangie** [comment]
OK, to make this more reproducable: 

1. Create a project in Zettlr with a document containing an undefined control sequence `\someexampletext`
2. Export the project to PDF
  - Watching the project directory, you should see that the tex2pdf working directory is created.
3. Attempt to export the project again without changing any files _within Zettlr_
  - I was changing across multiple files in VS Code and then attempting to re-export the project
  - No tex2pdf directory will be created, nothing happens.
4. Repeat 3, same result
5. Modify any file in the project within Zettlr (add and delete a single character, for example), and then export.
  - Zettlr exports the project successfully, even if export still fails (because we broke it)
  - Zettlr will not export the project until a file has been modified within the Edtior.

**Comment by nathanlesage** [comment]
> Once this bug is encountered, it occurs consistently until the application has been restarted - It will then work until the bug is encountered again.

This sounds as if there should be something logged to the log window (Develop menu -> Open logs)

**Comment by Kangie** [comment]
Now that I've been pointed at the logs I can give better bug reports!

I can't seem to replicate the behaviour at-will today (I spent a lot of effort fixing things yesterday and don't know at what point I stopped being able to trigger this), but I'm sure that I'll come across it again in the future. 

Closing the issue for now, because it's not reproducable and I don't have any useful logs. Will comment against it again when I'm able to reliably break export.

**Comment by Kangie** [comment]
OK, managed to reproduce this issue, at least enough to provide some logs.

Output to the log is: 
```
[11:25:35]	>>> IPC IN: dir-project-export
hash: 1113410143
[11:25:35]	Trying to run command through Application: dir-project-export
[11:25:35]	Unknown command received: dir-project-export
```

As expected, after restarting Zettlr this works perfectly.

I can't work out what causes Zettlr to do this, but I always encounter it after a failed export.

If I break my export template by inserting an undefined control sequence e.g.:

```
Error producing PDF.
! Undefined control sequence.
l.17 \breakpandoc
```

This issue sometimes occurs, but other times the next (non-broken) export is successful.

**Comment by Kangie** [comment]
Can confirm that this still occurs with the lastest (alpha.3) - not that I expected a fix as the ticket is still open!

**Comment by nathanlesage** [comment]
Alright, let me break this beauty a few times, and test this out. But seems as if there's an error in the command.

If the logs tell you "trying to run .." and then "no command registered" this rather means that there was an error during command run, which is one of the most annoying mistakes I made while coding that, because this way no error message is logged 🙄

One more thing: how does the project export look like? The more data you provide the closer I can narrow the circle to reproduce this error!

**Comment by Kangie** [comment]
> One more thing: how does the project export look like? The more data you provide the closer I can narrow the circle to reproduce this error!

Not sure exactly what mean here. The project will successfully export using my templates, however:

* If unescaped latex characters exist in my MarkDown source, this issue can be triggered (i.e. the template does not need to be broken, just a failed export using Pandoc).
* If I am working on a template and try to export, but there's an issue with the template syntax (causing the export to fail), this issue will occur.

The issue is not consistent, however a failed export *almost* always triggers this bug for me, so I've taken to restarting Zettlr when it happens. 

I really need to write up an external export engine that mirrors Zettlr's behaviour for quick tests like this / automation, but that's another matter entirely...

The 'on-disk' structure of my projects generally follows the below (Apologies for the censorship, work data...): 
![Zettlr-Project](https://user-images.githubusercontent.com/9733178/96836501-c9e4e700-1490-11eb-9a5c-24f952095586.png)

Noting that the project in question starts at 'Product 2\User Guide'

Basically laid out like that. There are subfolders under others, but they're laid out in the same way.

Let me know if I've missed what you were asking, happy to provide more info to track this one down.

I don't believe that this is related to project content (I've encountered it across 3 large projects), all indications seem to be that after a failed project export Zettlr forgets how to call pandoc.

**Comment by nathanlesage** [comment]
Still happening?

**Comment by Kangie** [comment]
Honestly not sure. Happy to close this for now and I'll get back to you if and when the export fails again. Hopefully if it does occur again the logs will provide more useful information!
